low-level/framework/platforms/server/error.py

```diff
@@ -0,0 +1,36 @@
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com
+
+class InterfaceError(Exception):
+    """Error Handling for Physical Interface/Ports Implementations."""
+
+    def __init__(self, rc, message, *args):
+        """Initialize the error information."""
+        self._rc = rc
+        self._desc = message % (args)
+
+    @property
+    def rc(self):
+        return self._rc
```

Comment:
why exactly keeping rc & desc private ? and then having member functions to access those !?

Reply:
Hi, I actually had zero thoughts about this, I simply copied from cortx-utils 😅. 
But, when I searched for it, it is for some name mangling purpose so that the value will not be accidentally overwritten by the class inheriting the original class, but I don't see any use-case here. Should I make it public?

Reply:
keeping it for now.